resources/src/asset.rs

```diff
@@ -0,0 +1,99 @@
+use utils;
+/// Permissions around who can own an asset and whether or
+/// not the asset issuer can freeze the asset.
+#[derive(Serialize, Deserialize, Debug)]
+struct Flag {
+    auth_required: bool,
+    auth_revocable: bool,
+}
+
+/// Assets are the units that are traded on the Stellar Network.
+/// An asset consists of an type, code, and issuer.
+/// Any asset can be traded for any other asset.
+/// https://www.stellar.org/developers/horizon/reference/resources/asset.html
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Asset {
+    asset_type: String,
+    asset_code: String,
+    asset_issuer: String,
+    #[serde(deserialize_with = "utils::deserialize_from_str")] amount: f64,
+    num_accounts: u32,
+    flags: Flag,
+}
+
+impl Asset {
+    ///   The type of this asset: “credit_alphanum4”, or “credit_alphanum12”.
+    ///   Returns a slice that lives as long as the asset does.
+    pub fn asset_type<'a>(&'a self) -> &'a str {
+        &(self.asset_type)
+    }
+
+    /// The code of this asset.
+    /// Returns a slice that lives as long as the asset does.
+    pub fn asset_code<'a>(&'a self) -> &'a str {
+        &(self.asset_code)
+    }
+
+    /// The issuer of this asset.  This corresponds to the id of an account.
+    /// Returns a slice that lives as long as the asset does.
+    pub fn asset_issuer<'a>(&'a self) -> &'a str {
+        &(self.asset_issuer)
+    }
+
+    /// The number of units of credit issued for this asset.
+    /// This number is scaled by 10 million to display the number if the format a
+    /// user would expect it in
+    /// https://www.stellar.org/developers/guides/concepts/assets.html
+    /// Returns a signed 64-bit integer.
+    pub fn amount(&self) -> i64 {
+        (self.amount * 10_000_000.0) as i64
+    }
+
+    /// The number of accounts that: 1) trust this asset and 2) where if the asset has the
+    /// auth_required flag then the account is authorized to hold the asset.
+    /// Returns an unsigned 32-bit integer
+    pub fn num_accounts(&self) -> u32 {
+        self.num_accounts
+    }
+
+    /// If this field is true it means the anchor must approve anyone who wants to
+    /// hold its credit, allowing it to control who its customers are
+    /// Returns a bool.
+    pub fn auth_required(&self) -> bool {
```

Comment:
I like starting with a helping verb in rust since `?` is not valid in an identifier:

```rust
pub fn is_auth_required(&self) -> bool {
```